### PR TITLE
Removed a few unused imports, missing references, and other errors

### PR DESCRIPTION
**Pull request by srri**
Decided to go through a bunch of the files to learn more about each, in the process also decided to clean things up where possible.

### CODE REVIEW

File: botocore/client.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 import botocore.serialize
 import botocore.validate
+import botocore.parsers
```
Comment (srri) [inline botocore/client.py]: Added as `botocore.parsers` is referenced further down in:
`response_parser = botocore.parsers.create_parser(protocol)`
but was not defined as it was never imported.

File: botocore/credentials.py [hunk 2]
```diff
@@ -1001,9 +1001,9 @@ def insert_before(self, name, credential_provider):
             & ordering can be discovered via ``self.available_methods``.
         :type name: string
 
-        :param cred_instance: An instance of the new ``Credentials`` object
+        :param credential_provider: An instance of the new ``Credentials`` object
```
Comment (srri) [inline botocore/credentials.py]: Fixing a few docstrings

File: botocore/endpoint.py [hunk 3]
```diff
@@ -23,7 +23,6 @@
 from botocore.vendored import six
 
 from botocore.awsrequest import create_request_object
-from botocore.exceptions import UnknownEndpointError
```
Comment (srri) [inline botocore/endpoint.py]: Unused reference

File: botocore/hooks.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 # language governing permissions and limitations under the License.
 import copy
 import logging
-from collections import defaultdict, deque, namedtuple
```
Comment (srri) [inline botocore/hooks.py]: Unused reference

File: botocore/hooks.py [hunk 5]
```diff
@@ -191,7 +191,6 @@ def _emit(self, event_name, kwargs, stop_on_response=False):
         :return: List of (handler, response) tuples from all processed
                  handlers.
         """
-        responses = []
```
Comment (srri) [inline botocore/hooks.py]: Unused variable, recreated further down anyways

File: botocore/parsers.py [hunk 6]
```diff
@@ -732,7 +732,7 @@ def _do_error_parse(self, response, shape):
                 return self._parse_error_from_body(response)
             except ResponseParserError as e:
                 LOG.debug(
-                    'Exception caught when parsing error response body:',
+                    'Exception caught when parsing error response body: {}'.format(e),
```
Comment (srri) [inline botocore/parsers.py]: Putting `e` to use

File: botocore/response.py [hunk 7]
```diff
@@ -12,13 +12,9 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-import sys
-import xml.etree.cElementTree
 import logging
 
-from botocore import ScalarTypes
```
Comment (srri) [inline botocore/response.py]: Unused imports

File: botocore/retryhandler.py [hunk 8]
```diff
@@ -140,7 +140,7 @@ def _create_single_response_checker(response):
         checker = CRC32Checker(header=response['crc32body'])
     else:
         # TODO: send a signal.
-        raise ValueError("Unknown retry policy: %s" % config)
+        raise ValueError("Unknown retry policy")
```
Comment (srri) [inline botocore/retryhandler.py]: `config` was undefined in this scope, not sure what it should be replaced with

File: botocore/signers.py [hunk 9]
```diff
@@ -22,7 +22,7 @@
 from botocore.exceptions import UnknownSignatureVersionError
 from botocore.exceptions import UnknownClientMethodError
 from botocore.exceptions import UnsupportedSignatureVersionError
-from botocore.utils import fix_s3_host, datetime2timestamp
```
Comment (srri) [inline botocore/signers.py]: Unused import

**Comment by jamesls** [comment]
Could you split the commits up?  I prefer not to have commits that are doing multiple things.  Your comments on this PR would be a good way to break up the commits :)

Have you had a chance to look at why codecov is complaining?

**Comment by srri** [comment]
Sure I can split it up. as for why codecov is complaining, im not sure. Nothing is changing that I can see that would affect coverage.

**Comment by kyleknap** [comment]
codecov is complaining I believe because you are overall removing lines and as a result a removal of lines that were completely covered would cause the overall code coverage to decrease. Let us know when you split up the commits.

**Pull request closed without merging**